tests/test_management.py

```diff
@@ -0,0 +1,240 @@
+            call_command('transmit_learner_data', api_user='bob')
+
+    def test_enterprise_customer_not_found(self):
+        faker = FakerFactory.create()
+        invalid_customer_id = faker.uuid4()
+        error = 'Enterprise customer {} not found, or not active.'.format(invalid_customer_id)
+        with raises(CommandError, message=error):
+            call_command('transmit_learner_data',
+                         api_user=self.api_user.username,
+                         enterprise_customer=invalid_customer_id)
+
+    def test_invalid_integrated_channel(self):
+        channel_code = 'abc'
+        error = 'Invalid integrated channel: {}'.format(channel_code)
+        with raises(CommandError, message=error):
+            call_command('transmit_learner_data',
+                         api_user=self.api_user.username,
+                         enterprise_customer=self.enterprise_customer.uuid,
+                         channel=channel_code)
+
+
+@responses.activate
+@mark.django_db
+@mark.parametrize('command_args,certificate,self_paced,end_date,passed,expected_completion', [
+    # Certificate marks course completion
+    (dict(), MOCK_CERTIFICATE, False, None, False, COMPLETE_CERTIFICATE),
+    # channel code is case-insensitive
+    (dict(channel='sap'), MOCK_CERTIFICATE, False, None, False, COMPLETE_CERTIFICATE),
+    (dict(channel='SAP'), MOCK_CERTIFICATE, False, None, False, COMPLETE_CERTIFICATE),
+    # enterprise_customer UUID gets filled in below
+    (dict(enterprise_customer=None), MOCK_CERTIFICATE, False, None, False, COMPLETE_CERTIFICATE),
+    (dict(enterprise_customer=None, channel='sap'), MOCK_CERTIFICATE, False, None, False, COMPLETE_CERTIFICATE),
+
+    # Instructor-paced course with no certificates issued yet results in incomplete course data
+    (dict(), None, False, None, False, dict(completed='false', timestamp='null', grade='null')),
+
+    # Self-paced course with no end date send grade=Pass, or grade=In Progress, depending on current grade.
+    (dict(), None, True, None, False, dict(completed='false', timestamp='null', grade='"In Progress"')),
+    (dict(), None, True, None, True, dict(completed='true', timestamp=NOW_TIMESTAMP, grade='"Pass"')),
+
+    # Self-paced course with future end date sends grade=Pass, or grade=In Progress, depending on current grade.
+    (dict(), None, True, FUTURE, False, dict(completed='false', timestamp='null', grade='"In Progress"')),
+    (dict(), None, True, FUTURE, True, dict(completed='true', timestamp=NOW_TIMESTAMP, grade='"Pass"')),
+
+    # Self-paced course with past end date sends grade=Pass, or grade=Fail, depending on current grade.
+    (dict(), None, True, PAST, False, dict(completed='true', timestamp=PAST_TIMESTAMP, grade='"Fail"')),
+    (dict(), None, True, PAST, True, dict(completed='true', timestamp=PAST_TIMESTAMP, grade='"Pass"')),
+])
+def test_transmit_learner_data(caplog, command_args, certificate, self_paced, end_date, passed, expected_completion):
```

Comment:
@pomegranited I'd suggest to either split this test into smaller chunks, or at least extract setup methods (`responses.add(...)`) - it's like 100 lines long, so it is hard to comprehend it.